src/admin/python/lsst/qserv/admin/cli/entrypoint.py

```diff
@@ -596,39 +559,23 @@ def worker_cmsd(
 @targs_options()
 @cmd_options()
 @options_file_option()
-def worker_xrootd(
-    ctx: click.Context,
-    debug_port: Optional[int],
-    db_uri: str,
-    db_admin_uri: str,
-    vnid_config: str,
-    cmsd_manager_name: str,
-    cmsd_manager_count: int,
-    repl_ctl_dn: str,
-    mysql_monitor_password: str,
-    db_qserv_user: str,
-    cmsd_worker_cfg_file: str,
-    cmsd_worker_cfg_path: str,
-    xrdssi_cfg_file: str,
-    xrdssi_cfg_path: str,
-    targs: Dict[str, str],
-    targs_file: str,
-    cmd: str,
-) -> None:
+def worker_xrootd(ctx: click.Context, **kwargs: Any) -> None:
+    targs = utils.targs(ctx)
+    targs = render(targs)
     script.enter_worker_xrootd(
-        targs=utils.targs(ctx),
-        debug_port=debug_port,
-        db_uri=db_uri,
-        db_admin_uri=db_admin_uri,
-        vnid_config=vnid_config,
-        repl_ctl_dn=repl_ctl_dn,
-        mysql_monitor_password=mysql_monitor_password,
-        db_qserv_user=db_qserv_user,
-        cmsd_worker_cfg_file=cmsd_worker_cfg_file,
-        cmsd_worker_cfg_path=cmsd_worker_cfg_path,
-        xrdssi_cfg_file=xrdssi_cfg_file,
-        xrdssi_cfg_path=xrdssi_cfg_path,
-        cmd=cmd,
+        targs=targs,
+        debug_port=targs["debug_port"],
+        db_uri=targs["db_uri"],
+        db_admin_uri=targs["db_admin_uri"],
+        vnid_config=targs["vnid_config"],
+        repl_ctl_dn=targs["repl_ctl_dn"],
+        mysql_monitor_password=targs["mysql_monitor_password"],
+        db_qserv_user=targs["db_qserv_user"],
+        cmsd_worker_cfg_file=targs["cmsd_worker_cfg_file"],
+        cmsd_worker_cfg_path=targs["cmsd_worker_cfg_path"],
+        xrdssi_cfg_file=targs["xrdssi_cfg_file"],
+        xrdssi_cfg_path=targs["xrdssi_cfg_path"],
+        cmd=targs["cmd"],
     )
```

Comment:
It doesn't pass in _each_ element of the targs dict though, it passes in _some_ elements. It passes in those elements because they are in the signature of the called function. `targs` is passed because the function does template expansion on config files (it does say so in the documentation of `enter_worker_xrootd`), and all CLI args (which is where targs comes from) are to be available for template expansion.